mypyc/emitmodule.py

```diff
@@ -63,57 +74,122 @@ def __init__(self, declaration: HeaderDeclaration, mark: bool) -> None:
         self.mark = False
 
 
+class MypycPlugin(Plugin):
```

Comment:
Maybe move this to a separate module, since a mypy plugin seems like its own thing.

Reply:
The plugin is pretty closely tied into other logic in this module, since what it does is enforce the caching and group settings from the rest of the driver logic, so I would prefer to keep it in the same module as that code. (Splitting it out in the most obvious way would introduce a cycle, though that is avoidable without much pain.)

I do think that it will be worth splitting and maybe renaming `emitmodule` to separate `GroupGenerator` from the rest. My inclination is to do that in a follow-up PR to keep down the diff noise for this diff and to avoid needing to rebase some follow-up code I've already written, though I could do it now too if you think it's important.

Reply:
Sounds reasonable.